soh/soh/Enhancements/randomizer/draw.cpp

```diff
@@ -108,19 +149,25 @@ extern "C" void Randomizer_DrawCompass(PlayState* play, GetItemEntry* getItemEnt
 }
 
 extern "C" void Randomizer_DrawBossKey(PlayState* play, GetItemEntry* getItemEntry) {
-    s8 keysCanBeOutsideDungeon = getItemEntry->getItemId == RG_GANONS_CASTLE_BOSS_KEY ?
-        DUNGEON_ITEMS_CAN_BE_OUTSIDE_DUNGEON(RSK_GANONS_BOSS_KEY) :
-        DUNGEON_ITEMS_CAN_BE_OUTSIDE_DUNGEON(RSK_BOSS_KEYSANITY);
-    s8 isColoredKeysEnabled = keysCanBeOutsideDungeon && CVarGetInteger(CVAR_RANDOMIZER_ENHANCEMENT("MatchKeyColors"), 1);
-    s16 color_slot;
-    color_slot = getItemEntry->getItemId - RG_FOREST_TEMPLE_BOSS_KEY;
-    s16 colors[6][3] = {
-        { 4, 195, 46 },   // Forest Temple
-        { 237, 95, 95 },  // Fire Temple
-        { 85, 180, 223 }, // Water Temple
-        { 222, 158, 47 }, // Spirit Temple
-        { 126, 16, 177 }, // Shadow Temple
-        { 210, 0, 0 }     // Ganon's Castle
+    s8 isCustomKeysEnabled = CVarGetInteger(CVAR_RANDOMIZER_ENHANCEMENT("CustomKeyModels"), 1);
+    s16 slot = getItemEntry->getItemId - RG_FOREST_TEMPLE_BOSS_KEY;
+
+    std::string CvarValue[6] = {
+        "Forest",
+        "Fire",
+        "Water",
+        "Spirit",
+        "Shadow",
+        "Ganons",
+    };
```

Comment:
A little weird that here only the different parts are in the array instead of the whole cvar, should probably use the same for both, not really important which.

Reply:
I did this to avoid having an ever expanding mess of tables, but am not happy with this layout due to issues with CVAR_COSMETIC. The string manipulation needed here to make the cvars clean is obnoxious as the macro does not like me feeding in manipulated strings for whatever reason, so I am accepting a bad solution to see if better ones arrive or at the very least, to not get bogged down here.